src/arch/xtensa/drivers/timer.c

```diff
@@ -103,7 +103,7 @@ uint64_t arch_timer_get_system(struct timer *timer)
 	return time;
 }
 
-int arch_timer_set(struct timer *timer, uint64_t ticks)
+uint64_t arch_timer_set(struct timer *timer, uint64_t ticks)
```

Comment:
This function also returns negative values (errorcodes).

Reply:
ahh right I overlooked that.